Issue 5752 dark link

elzadj opened the pull request:
# Description

Fixed and migrates the wrong link colour in the default SC. 

Fixes #5752 

# How Has This Been Tested?

1. Test on an old SC from master. If Maxi Default is active, it should change the colours on the next frontend load, no need to do anything in the editor. 
2. Reset the SCs and test again. 

```wp.data.dispatch('maxiBlocks/style-cards').resetSC()``` 

The colours should be ok too.
# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a migration process for updating style card link palette colors.
	- Added a method to run the migration automatically during initialization.

- **Bug Fixes**
	- Updated link palette color values from `5` to `4` across the application for consistent styling.

- **Style**
	- Minor formatting adjustments made for improved code readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

elzadj committed:
Fix links colour for new installations

elzadj committed:
Add php migrator for maxi default

elzadj committed:
Add one-time link colour migrator if Maxi default sc  is active

Olekrut reviewed (changes requested):
https://jam.dev/c/6e561ad6-ab79-450b-9fd1-a28f71264712
were overwritten colour for all links on frontend after reset

elzadj committed:
Change the migration trigger

elzadj committed:
fix links after the reset

elzadj committed:
Fix default sc_string

elzadj commented:
> https://jam.dev/c/6e561ad6-ab79-450b-9fd1-a28f71264712 were overwritten colour for all links on frontend after reset

@Olekrut 
Fixed, please do this:
1. Pull this branch changes
2. Switch to master
3. Reset SCs
4. Remove maxi_blocks_link_palette_color_migrated option from your local _options DB table (this is for the one-time migrator: you already run it, so we need to remove it)
5. Switch to this branch
6. Visit any admin page on the site to trigger the migrator (I moved it from the frontend trigger to not run it each time on each page)
7. Check frontend
8. Reset SCs
9. Check frontend
